package.json

```diff
@@ -193,6 +193,7 @@
     "vscode": "^1.17.0"
   },
   "activationEvents": [
+    "onDebug",
```

Comment:
Does this mean that the C# extension will activate whenever the user starts debugging any non-CS file? If so, wouldn't that be a bad experience if the user installs the C# extension, tries to debug JS and the C# extension starts downloading OmniSharp and the debugger?

Reply:
This activation event is required for DebugConfigurationProviders. 

https://code.visualstudio.com/updates/v1_17#_debug-contributions-in-packagejson

Reply:
Sure, but that's not what I asked. :smile:

Reply:
Sorry, I forgot to mention that I will still look into if the C# extension is activated in a node project. That was an FYI of why its there. :)

Reply:
Thanks!

Reply:
@DustinCampbell You are correct. It will activate on any F5 on any file type. 😞 

![image](https://user-images.githubusercontent.com/3953714/32348198-42b1f8e4-bfd0-11e7-897d-6ad393f68184.png)

Reply:
That's what I guessed. @gregg-miskelly: What do you think about this? I'm not sure there's much we can do about it, but I'm betting we'll hear about this.

Reply:
It certainly isn't ideal. I am assuming VS Code will activate us only once is a given VS Code session, so there isn't much we could do about this by trying to filter the activations. Right?

Reply:
I think that's right. AFAIK, we'd essentially need to test all of our activation events manually when the extension activates.

Reply:
Do you think there is anything we should improve for 1.13? 

Here is one idea I had, but I am not sure if I like it - we could ship with an activation handler of '*' (see [docs](https://code.visualstudio.com/docs/extensionAPI/activation-events#_activationevents)). This will make sure that our extension activates right after the user installs the extension so hopefully it will feel connected with installing it. Then in our activate handler we update the package.json to the set of things currently in there.

Reply:
That's not a bad idea, though you're right that it's pretty gross. :smile: Do you know if VS Code caches activation events from installed extensions?

Personally, I'm OK with living with this for 1.13 and then improve it later.

Reply:
The C++ extension rewrites several section of their package.json, but I am not sure if this is one of them. @WardenGnaw @pieandcakes do you know?

Reply:
@DustinCampbell VsCode rereads the package.json to see what the activation events are. If we originally had ```*``` and replaced it with something else. On the next session of VSCode, it will reread the activationEvents array.

The C++ extension currently does this.

Reply:
Clever! In that case, this seems like a reasonable approach.

Reply:
@gregg-miskelly: Do you think we should do this for 1.13, or are you OK with doing it in 1.14, Same question for @TheRealPiotrP

Reply:
@DustinCampbell do you mean the change all up or the switch to use the '*' activation event?

If change all up - @WardenGnaw how broken are we in the VS Code 1.18 (the one that is just about to come out in stable channel) without this? Are the things we are replacing just deprecated or are the broken?

If the '*' activation - I don't have a strong opinion. I am happy to wait for 1.14 on that part if you would like.

Reply:
I meant using the '*' activation event. I'm OK with waiting for 1.14 for this.

Reply:
Without the '*' activation event, we are not broken. 

This was to fix the rare scenarios of people installing the C# extension, not opening a C# project for a while, then going to a non-C# project and then F5'ing would case the C# dependencies to start downloading.